### DISCUSSION TITLE
UE environment on local machine +Deep learning model on remote machine

### DISCUSSION THREAD

**Discussion by dreamDasci** [microsoft/AirSim#4117] [Support Q&A]
Hi, I have a windows pc where I'm able to run a (.exe) file that simulates an Unreal Engine environment. I've written a deep learning model that predicts the next action. This model needs the current position of the drone as well as the camera image.
I want to run this model on a **remote ubuntu machine**.
I'm able to **ssh** the remote machine and I can do **local and remote port forwarding (tunnels)**.
I want a way to **give** my model the **inputs** it needs and to **control** the drone via AirSim in the environment.
Does anyone have an idea about what I can do to achieve that?
Currently, when I execute my Python script(on remote machine), there's an error when running the 'client.confirmConnection()' command. 

`self._error msgpackrpc.error.TransportError: Retry connection over the limit `

My code is largely based on **https://github.com/aqeelanwar/PEDRA**.
Besides the Deep learning models, it gives a (.exe) UE simulated environments.

**Comment by rajat2004** [comment]
You're able to ping the machines in both directions correct? Make sure to set the `ip` and `port` when [creating the VehicleClient](https://microsoft.github.io/AirSim/api_docs/html/#airsim.client.VehicleClient)
